geode-lucene/src/test/java/org/apache/geode/cache/lucene/internal/cli/LuceneIndexCommandsDUnitTest.java

```diff
@@ -154,24 +169,42 @@ public void listIndexWithStatsShouldReturnCorrectStats() throws Exception {
 
 
   @Test
-  public void createIndexShouldCreateANewIndex() throws Exception {
+  @Parameters(method = "getRegionCreationOrder")
+  public void createIndexShouldCreateANewIndex(RegionCreationOrder regionCreationOrder)
+      throws Exception {
     CommandStringBuilder csb = new CommandStringBuilder(LuceneCliStrings.LUCENE_CREATE_INDEX);
     csb.addOption(LuceneCliStrings.LUCENE__INDEX_NAME, INDEX_NAME);
     csb.addOption(LuceneCliStrings.LUCENE__REGION_PATH, REGION_NAME);
     csb.addOption(LuceneCliStrings.LUCENE_CREATE_INDEX__FIELD, "field1,field2,field3");
 
-    gfsh.executeAndAssertThat(csb.toString()).statusIsSuccess();
+    createAndVerifyLuceneIndexAndRegion(regionCreationOrder, csb);
 
     serverVM.invoke(() -> {
       LuceneService luceneService = LuceneServiceProvider.get(getCache());
-      createRegion();
       final LuceneIndex index = luceneService.getIndex(INDEX_NAME, REGION_NAME);
       assertArrayEquals(new String[] {"field1", "field2", "field3"}, index.getFieldNames());
     });
   }
 
+  private void createAndVerifyLuceneIndexAndRegion(RegionCreationOrder regionCreationOrder,
+      CommandStringBuilder csb) {
+    if (regionCreationOrder.equals(RegionCreationOrder.REGION_CREATED_BEFORE_LUCENE_INDEX)) {
+      serverVM.invoke(() -> LuceneServiceImpl.LUCENE_REINDEX = true);
+      serverVM.invoke(() -> createRegion());
+      gfsh.executeAndAssertThat(csb.toString()).statusIsSuccess()
+          .containsOutput("Successfully created lucene index");
+    } else if (regionCreationOrder.equals(RegionCreationOrder.REGION_CREATED_AFTER_LUCENE_INDEX)) {
```

Comment:
You might want a LUCENE_REINDEX = true and still create the index before the region scenario